chore: remove Endpoint and Protocol types (#654)
0xPolygonMiden/miden-node: CHANGELOG.md

### Changes

- [BREAKING] Updated minimum Rust version to 1.84.

> - [BREAKING] Replace `Endpoint` and `Protocol` types with `url::Url` and update node TOML configuration to treat endpoints as a single string (#654).

Reviewer: Nit: changelog should be from the user/operator perspective (i.e. they don't know `url::Url` etc).

```suggestion
- [BREAKING] `Endpoint` configuration simplified to a single string (#654).
```
